docs/cryptodoc/src/07_rng.rst

```diff
@@ -1054,5 +1060,29 @@ It works as follows:
    4. if (:math:`min \leq r < max`) return ``r``.
    5. Go to Step 3.
 
-**Conclusion:** The algorithm is a slight adaptation of Algorithm B.1 in Section B.4 of [TR-02102-1]_
+**Note:** Technically, the base implementation above should be defined for exactly :math:`min = 0`
+not :math:`min \in \{0,1\}`. When `adding this special case in Botan 3.1.0 <https://github.com/randombit/botan/pull/3594>`_
+this was done to avoid breaking existing test cases that depended on the determinism of this method for :math:`min = 1`.
+
+.. admonition:: ``BigInt::random_integer()`` (for :math:`min > 1`)
+
+   **Input:**
+
+   -  ``rng``: random number generator (see :ref:`rng/main` for available algorithms)
+   -  ``min``: integer lower bound of desired range (with :math:`min > 1`)
+   -  ``max``: integer upper bound of desired range, excluding ``max``
+
+   **Output:**
+
+   -  ``r``: :math:`min \leq r < max`
+
+   **Steps:**
+
+   1. Preliminary parameter requirement checks are conducted. ``min`` must be
+      smaller than ``max``, and neither can be negative.
+   2. Calculate ``diff`` as the difference between ``max`` and ``min``
+   3. Call ``BigInt::random_integer()`` for the interval ``0..diff`` to obtain ``n``
```

Comment:
in the interval `0..diff`: `diff` is excluded. Maybe we want to write something like:

Call ``BigInt::random_integer()`` to obtain a random number _n_ with _0 <= n < diff_

Reply:
Making it:

```
3. Call ``BigInt::random_integer(rng, 0, diff)`` to obtain :math:`n`
```

No need to overly abstract that. I code we trust.